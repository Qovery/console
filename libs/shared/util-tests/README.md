# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
