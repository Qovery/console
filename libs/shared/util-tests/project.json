{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
