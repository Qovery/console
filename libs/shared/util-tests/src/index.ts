export * from './lib/shared-util-tests'
