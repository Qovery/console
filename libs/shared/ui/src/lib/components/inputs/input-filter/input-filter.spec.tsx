import { renderWithProviders, screen } from '@qovery/shared/util-tests'
import InputFilter from './input-filter'

describe('InputFilter', () => {
  const options = [
    { label: 'Option 1', value: 'option1' },
    { label: 'Option 2', value: 'option2' },
    { label: 'Option 3', value: 'option3' },
  ]

  it('renders the button with the provided name', () => {
    renderWithProviders(<InputFilter name="Filter" nameKey="filter" options={options} onChange={jest.fn()} />)

    expect(screen.getByText('Filter')).toBeInTheDocument()
  })

  it('opens the select input on button click', async () => {
    const { userEvent } = renderWithProviders(
      <InputFilter name="Filter" nameKey="filter" options={options} onChange={jest.fn()} />
    )

    const button = screen.getByText('Filter')
    await userEvent.click(button)

    expect(screen.getByText('Filter')).toBeInTheDocument()
  })

  it('selects an option and triggers onChange callback', async () => {
    const handleChange = jest.fn()
    const { userEvent } = renderWithProviders(
      <InputFilter name="Filter" nameKey="filter" options={options} onChange={handleChange} />
    )

    const button = screen.getByText('Filter')
    await userEvent.click(button)

    const option2 = screen.getByText('Option 2')
    await userEvent.click(option2)

    expect(handleChange).toHaveBeenCalledWith('filter', 'option2')
    expect(screen.getByTestId('clear-btn')).toBeInTheDocument()
  })

  it('clears the selection on clear button click', async () => {
    const handleChange = jest.fn()
    const { userEvent } = renderWithProviders(
      <InputFilter name="Filter" nameKey="filter" options={options} onChange={handleChange} defaultValue="option1" />
    )

    const button = screen.getByText('Option 1')
    await userEvent.click(button)

    const clearButton = screen.getByTestId('clear-btn')
    await userEvent.click(clearButton)

    expect(handleChange).toHaveBeenCalledWith('filter', undefined)
    expect(screen.queryByTestId('clear-btn')).not.toBeInTheDocument()
  })

  it('should match input filter without value', () => {
    const handleChange = jest.fn()

    const { container } = renderWithProviders(
      <InputFilter name="Filter" nameKey="filter" options={options} onChange={handleChange} />
    )

    expect(container).toMatchSnapshot()
  })

  it('should match input filter with value', () => {
    const handleChange = jest.fn()

    const { container } = renderWithProviders(
      <InputFilter name="Filter" nameKey="filter" options={options} onChange={handleChange} defaultValue="option1" />
    )

    expect(container).toMatchSnapshot()
  })

  it('should match input filter with loading', () => {
    const handleChange = jest.fn()

    const { container } = renderWithProviders(
      <InputFilter
        name="Filter"
        nameKey="filter"
        options={options}
        onChange={handleChange}
        defaultValue="option1"
        isLoading={true}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
