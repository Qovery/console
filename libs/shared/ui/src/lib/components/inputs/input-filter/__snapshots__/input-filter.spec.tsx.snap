// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFilter should match input filter with loading 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
      type="button"
    >
      <div
        class="flex w-12 justify-center"
      >
        <div
          class="aspect-square border-2  rounded-full border-solid border-neutral-350 border-r-neutral-50 animate-spin w-4 "
          data-testid="spinner"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`InputFilter should match input filter with value 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
      type="button"
    >
      Option 1
      <span
        class="relative left-1 px-1 py-1"
        data-testid="clear-btn"
        role="button"
      >
        <i
          aria-hidden="true"
          class="fa-solid fa-xmark "
        />
      </span>
    </button>
  </div>
</div>
`;

exports[`InputFilter should match input filter without value 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400 gap-2"
      type="button"
    >
      Filter
      <i
        aria-hidden="true"
        class="fa-solid fa-magnifying-glass "
      />
    </button>
  </div>
</div>
`;
